zaza/openstack/charm_tests/cloudkitty/tests.py

```diff
@@ -0,0 +1,128 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Encapsulate Cloudkitty testing."""
+
+import logging
+
+import zaza.openstack.charm_tests.test_utils as test_utils
+import zaza.openstack.utilities.openstack as openstack_utils
+
+from cloudkittyclient import client
+
+
+class CloudkittyTest(test_utils.OpenStackBaseTest):
+    """Encapsulate Cloudkitty tests."""
+
+    API_VERSION = '1'
+
+    @classmethod
+    def setUpClass(cls):
+        """Run class setup for running Cloudkitty tests."""
+        super(CloudkittyTest, cls).setUpClass()
+        cls.current_release = openstack_utils.get_os_release()
+
+        logging.info('Instantiating cloudkitty client...')
+        cls.cloudkitty = client.Client(
+            CloudkittyTest.API_VERSION,
+            session=cls.keystone_session
+        )
+
+    def test_400_api_connection(self):
+        """Simple api calls to check service is up and responding."""
+        report = self.cloudkitty.report
+        tenants_list = report.get_tenants()
+        assert tenants_list == []
+
+    def test_401_module_enable_and_disable(self):
+        """Test enable and disable module via API."""
+        enable = True
+        disable = False
+
+        rating = self.cloudkitty.rating
+
+        logging.info('Enabling hashmap module.')
+        rating.update_module(module_id='hashmap', enabled=enable)
+
+        module = rating.get_module(module_id='hashmap')
+        assert module.get('enabled') == enable
+
+        logging.info('Disabling hashmap module.')
+        rating.update_module(module_id='hashmap', enabled=disable)
+
+        module = rating.get_module(module_id='hashmap')
+        assert module.get('enabled') == disable
+
+    def test_402_service_create_and_delete(self):
+        """Test service create and delete via API."""
+        rating = self.cloudkitty.rating
+        rating.update_module(module_id='hashmap', enabled=True)
+
+        hashmap = self.cloudkitty.rating.hashmap
+        service = hashmap.create_service(name='test')
+
+        assert service.get('name') == 'test'
+
+        service_id = service.get('service_id')
+        hashmap.delete_service(service_id=service_id)
+
+    def test_403_field_create_and_delete(self):
+        """Test field create and delete via API."""
+        rating = self.cloudkitty.rating
+        rating.update_module(module_id='hashmap', enabled=True)
+
+        hashmap = self.cloudkitty.rating.hashmap
+        service = hashmap.create_service(name='test')
+
+        service_id = service.get('service_id')
+        field = hashmap.create_field(name='test', service_id=service_id)
+
+        field_id = field.get('field_id')
+        hashmap.delete_field(field_id=field_id)
+        hashmap.delete_service(service_id=service_id)
+
+    def test_404_mapping_create_and_delete(self):
```

Comment:
This test actually includes test 401,402 and 403. They are not needed if we are testing those components in the test here, so I would keep only test 404.